js/graphql.md

```diff
@@ -0,0 +1,1374 @@
+terminal window. If now error messages are thrown then you are good to go
+and can deploy your new API.
+
+```bash
+amplify push
+```
+
+Go to AWS CloudFormation to view it. You can also find your project assets in the amplify/backend folder under your API.
+
+Once the API is finsihed deploying, try going to the AWS AppSync console and
+running some of these queries in your new API's query page.
+
+```
+# Create a blog. Remember the returned id.
+# Provide the returned id as the "blogId" variable.
+mutation CreateBlog {
+  createBlog(input: {
+    name: "My New Blog!"
+  }) {
+    id
+    name
+  }
+}
+
+# Create a post and associate it with the blog via the "postBlogId" input field.
+# Provide the returned id as the "postId" variable.
+mutation CreatePost($blogId:ID!) {
+  createPost(input:{title:"My Post!", postBlogId: $blogId}) {
+    id
+    title
+    blog {
+      id
+      name
+    }
+  }
+}
+
+# Create a comment and associate it with the post via the "commentPostId" input field.
+mutation CreateComment($postId:ID!) {
+  createComment(input:{content:"A comment!", commentPostId:$postId}) {
+    id
+    content
+    post {
+      id
+      title
+      blog {
+        id
+        name
+      }
+    }
+  }
+}
+
+# Get a blog, its posts, and its posts comments.
+query GetBlog($blogId:ID!) {
+  getBlog(id:$blogId) {
+    id
+    name
+    posts(filter: {
+      title: {
+        eq: "My Post!"
+      }
+    }) {
+      items {
+        id
+        title
+        comments {
+          items {
+            id
+            content
+          }
+        }
+      }
+    }
+  }
+}
+
+# List all blogs, their posts, and their posts comments.
+query ListBlogs {
+  listBlogs { # Try adding: listBlog(filter: { name: { eq: "My New Blog!" } })
+    items {
+      id
+      name
+      posts { # or try adding: posts(filter: { title: { eq: "My Post!" } })
+        items {
+          id
+          title
+          comments { # and so on ...
+            items {
+              id
+              content
+            }
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+If you want to update your API, open your project's `backend/api/~apiname~/schema.graphql` file (NOT the one in the `backend/api/~apiname~/build` folder) and edit it in your favorite code editor. You can compile the `backend/api/~apiname~/schema.graphql` by running:
+
+```
+amplify api gql-compile
+```
+
+and view the compiled schema output in `backend/api/~apiname~/build/schema.graphql`.
+
+You can then push updated changes with:
+
+```
+amplify push
+```
+
+## Directives
+
+### @model
+
+Object types that are annotated with `@model` are top-level entities in the
+generated API. Objects annotated with `@model` are stored in DynamoDB and are
+capable of being protected via `@auth`, related to other objects via `@connection`,
+and streamed into Elasticsearch via `@searchable`.
+
+#### Definition
+
+```
+directive @model(
```

Comment:
I think it might be worthwhile calling this "API Definition" and then explaining with a one-liner "XXX below defines the `@model` capabilities, allowing you to YYYY...."

Reason I say this is some people have confused this in the older docs with what they blindly copied & pasted into their SDL, which is actually in the "Usage" section below.